### PR TITLE
Add UNSPECIFIED to every enum

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
`UNSPECIFIED` value was added to every enum as 0 item.

**Why?**
To follow [protobuf style guide](https://developers.google.com/protocol-buffers/docs/style#enums) and to be compatible with [buf linter](https://buf.build/docs/lint-overview).

**How did you test it?**
Run all tests.

**Potential risks**
There is a risk that some default values are not set correctly and `UNSPECIFIED` value can go through the code and stored in DB.

### CODE REVIEW

File: common/enums/defaults.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package enums
```
Comment (alexshtin) [inline common/enums/defaults.go:25]: All enums in this file are considered to be optional. If value is not set (i.e. set to `UNSPECIFIED`) some default value will be used.

File: proto/common/server_enum.proto [hunk 2]
```diff
@@ -25,44 +25,48 @@ package common;
...
+    // Task produced by history service.
+    TASK_SOURCE_HISTORY = 1;
+    // Task produced from matching db backlog.
+    TASK_SOURCE_DB_BACKLOG = 2;
 }
 
 enum TaskCategory {
     TASK_CATEGORY_UNSPECIFIED = 0;
-    // Transfer is the task type for transfer task
-    TASK_CATEGORY_TRANSFER = 2; // starting from 2 here to be consistent with the row type define for cassandra
```
Comment (alexshtin) [inline proto/common/server_enum.proto:42]: I wasn't able to figure out why it started with 2 and reset it to normal 1.

File: service/history/mutableStateTaskGenerator.go [hunk 3]
```diff
@@ -205,7 +205,7 @@ func (r *mutableStateTaskGeneratorImpl) generateDelayedDecisionTasks(
 		commonpb.CONTINUE_AS_NEW_INITIATOR_DECIDER:
 		firstDecisionDelayType = persistence.WorkflowBackoffTimeoutTypeCron
 	default:
-		return serviceerror.NewInternal(fmt.Sprintf("unknown iterator retry policy: %v", startAttr.GetInitiator()))
```
Comment (alexshtin) [inline service/history/mutableStateTaskGenerator.go:208]: :facepalm:

File: common/enums/defaults.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+func SetDefaultTasklListType(f *tasklistpb.TaskListType){
+	if *f == tasklistpb.TASK_LIST_TYPE_UNSPECIFIED{
+		*f = tasklistpb.TASK_LIST_TYPE_DECISION
+	}
+}
+
+func SetDefaultTasklListKind(f *tasklistpb.TaskListKind){
+	if *f == tasklistpb.TASK_LIST_KIND_UNSPECIFIED{
```
Comment (samarabbas) [inline common/enums/defaults.go:65]: Is this public API? I'm surprised if it is as client never cares about tasklist kind.  If this is internal then we should make it required.

File: proto/common/server_enum.proto [hunk 5]
```diff
@@ -25,44 +25,48 @@ package common;
...
-    // Timer is the task type for timer task
-    TASK_CATEGORY_TIMER = 3;
-    // Replication is the task type for replication task
-    TASK_CATEGORY_REPLICATION = 4;
+    // Transfer is the task type for transfer task.
+    TASK_CATEGORY_TRANSFER = 1;
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply proto/common/server_enum.proto:45]: This is an enum:
```
const (
	// Row types for table executions
	rowTypeShard = iota
	rowTypeExecution
	rowTypeTransferTask
	rowTypeTimerTask
	rowTypeReplicationTask
	rowTypeDLQ
)
```
There is shard info but it has value 0. And this proto enum doesn't intersect with go enum above in a code.

File: common/enums/defaults.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+func SetDefaultHistoryEventFilterType(f *filterpb.HistoryEventFilterType){
+	if *f == filterpb.HISTORY_EVENT_FILTER_TYPE_UNSPECIFIED{
+		*f = filterpb.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT
+	}
+}
+
+func SetDefaultTasklListType(f *tasklistpb.TaskListType){
+	if *f == tasklistpb.TASK_LIST_TYPE_UNSPECIFIED{
+		*f = tasklistpb.TASK_LIST_TYPE_DECISION
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply common/enums/defaults.go:60]: Made it required.

File: common/enums/defaults.go (hunk 4 above)
Comment (alexshtin) [reply common/enums/defaults.go:65]: Yes, it is used in `DescribeTaskList` and actually validated everywhere where `TaskList` is a part of request. Left it optional and set it to `NORMAL` if not set on request.

File: common/enums/defaults.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+func SetDefaultContinueAsNewInitiator(f *commonpb.ContinueAsNewInitiator){
+	if *f == commonpb.CONTINUE_AS_NEW_INITIATOR_UNSPECIFIED{
+		*f = commonpb.CONTINUE_AS_NEW_INITIATOR_DECIDER
+	}
+}
+
+func SetDefaultQueryConsistencyLevel(f *querypb.QueryConsistencyLevel){
+	if *f == querypb.QUERY_CONSISTENCY_LEVEL_UNSPECIFIED{
+		*f = querypb.QUERY_CONSISTENCY_LEVEL_EVENTUAL
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply common/enums/defaults.go:48]: I am setting this to `STRONG` for now, but Max suggested to remove this field completely from request. I will do it as separate PR.

**Pull request merged**